Desktop/widgets/boundcontrollavaantextarea.cpp

```diff
@@ -117,7 +118,7 @@ QString BoundControlLavaanTextArea::rScriptDoneHandler(const QString & result)
 
 	boundValue["columns"] = columns;
 
-	setBoundValue(boundValue);
+	setBoundValue(boundValue, !_control->form()->analysisObj()->wasUpgraded());
```

Comment:
Quite awkward here. If the analysis should not be run when wasUpgraded in true, then set this condition directly in the Analysis::boundValueChangedHandler() method.

Reply:
which condition?

Reply:
The condition 'wasUpgraded is true'

Reply:
But if I put it there then nothing is ever run, which means _wasUpgraded is never set to false... Which means it won't work.

Reply:
As in:
```
void Analysis::boundValueChangedHandler()
{
	incrementRevision(); // To make sure we always process all changed options we increment the revision whenever anything changes

	Log::log() << "Option changed for analysis '" << name() << "' and id " << id() << ", revision incremented to: " << _revision << std::endl;

	if (_refreshBlocked)
		return;

	if (form() && (form()->hasError() || !form()->runOnChange()))
		return;

	run();
}
```

So I could put that condition there, but it looks like it won't get to `run` then whenever `_wasUpgraded == true`.

But it is only set to `false` in `setResults` and `setVersion`, which means that it first needs to run, no?

Reply:
I mean, I totally get why you find the way I've written it now "awkward" but I thought it would be good to have the decision on whether rscripts should trigger valueChanged or not to be in the qml widget.

I could forward the `wasUpgraded()` function to `BoundControlBase` or something. That might make it look better at least.